src/NHibernate/Criterion/SQLCriterion.cs

```diff
@@ -42,7 +43,7 @@ public override SqlString ToSqlString(ICriteria criteria, ICriteriaQuery criteri
 					parameters[paramPos++].BackTrack = parameter.BackTrack;
 				}
 			}
-			return _sql.Replace("{alias}", criteriaQuery.GetSQLAlias(criteria));
+			return SQLProjection.GetSqlString(criteria, criteriaQuery, _sql);
```

Comment:
Slight code smell, but well, not worth introducing a common helper class for this. So let it be.

Reply:
This was copied from the other PR as is. The initial change is just a reset of my changes on top of master.